src/main/scala/org/constellation/http/Loadbalancer.scala

```diff
@@ -34,8 +35,27 @@ class Loadbalancer(port: Int = 9000, host: String = "localhost") {
 
   private val upstreamIterator = Ref.unsafe[IO, Iterator[Addr]](List.empty[Addr].iterator)
 
+  private val redirectToMaintenance: Ref[IO, Boolean] = Ref.unsafe(false)
+
   private val sessions = new SessionCache()
 
+  def shouldRedirectToMaintenance: IO[Boolean] =
+    redirectToMaintenance.get
+
+  def enableRedirectingToMaintenance: IO[Unit] =
+    redirectToMaintenance.set(true)
+
+  def disableRedirectingToMaintenance: IO[Unit] =
+    redirectToMaintenance.set(false)
+
+  def reset: IO[Unit] =
+    for {
+      _ <- upstream.set(List.empty[Addr])
```

Comment:
You should use modify instead of set. Set updates immediately, even if another update is in progress. Modify waits and doesn't make a modification when other part of the app tries to modify the value

Reply:
Yeah I know. I was just wondering if for that case we need to wait 🤔 .

I mean only one person triggers the rollback so we won't have situation that someone calls reset many times.

But I can change it anyway

Reply:
I'm not thinking about triggering that mode multiple times but what if there is an ongoing change in upstream 🤔